src/Support/StateIdentity.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Thunk\Verbs\Support;
+
+use InvalidArgumentException;
+use Thunk\Verbs\State;
+
+class StateIdentity
+{
+    public static function from(object $source): static
+    {
+        return match (true) {
+            $source instanceof self => $source,
+            $source instanceof State => new static(state_type: $source::class, state_id: $source->id),
+            default => static::fromGenericObject($source),
+        };
+    }
+
+    protected static function fromGenericObject(object $source): static
+    {
+        $state_id = data_get($source, 'state_id');
+        $state_type = data_get($source, 'state_type');
+
+        if (is_int($state_id) && is_string($state_type)) {
```

Comment:
```suggestion
        if (
            (
                \Illuminate\Support\Str::isUlid($state_id)
                || \Illuminate\Support\Str::isUuid($state_id)
                || is_int($state_id)
            )
            && is_string($state_type)
        ) {
```

This is formatted terribly, but I think we need to check for more than just an `int` for the `$state_id`